src/components/pages/PointSystem/TaskDialog.vue

```diff
@@ -0,0 +1,197 @@
+<template>
+  <dialog-base :visible.sync="isVisible" class="task-dialog" :title="pointsForCategory.titleProgress">
+    <div>
+      <p class="task-dialog__title-progress">
+        Complete {{ pointsForCategory.titleProgress }}-related tasks in order to level up your skill
+      </p>
+      <div class="task-dialog__card-progress">
+        <div class="current-level">
+          <p>Your level</p>
+          <p>
+            LVL {{ levelCurrent }} <span>/ LVL {{ maxLevel }}</span>
+          </p>
+        </div>
+        <div class="progress-container">
+          <div class="progress-bar" :style="{ width: progressPercentage + '%' }"></div>
+        </div>
+
+        <div class="next-level">
+          <p>Next level reward</p>
+          <p>{{ pointsForCategory.nextLevelRewardPoints }}</p>
+        </div>
+      </div>
+      <div class="task-dialog__card-current">
+        <div class="img-title">
+          <token-logo v-if="isTokenImage" :token="getImageSrc(imageName)" size="small" />
+          <img v-else :src="getImageSrc(imageName)" :alt="imageName" />
+          <p>{{ pointsForCategory.titleTask }}</p>
+        </div>
+        <p class="description">{{ pointsForCategory.descriptionTask }}</p>
+        <s-divider />
+        <p class="currently-amount">
+          Currently: <span> ${{ pointsForCategory.currentProgress.toFixed(2) }}</span>
+        </p>
+      </div>
+    </div>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import { mixins, components } from '@soramitsu/soraneo-wallet-web';
+import { Component, Mixins, Prop } from 'vue-property-decorator';
+
+import { getImageSrc, isTokenImage, MAX_LEVEL } from '@/consts/pointSystem';
+import { CalculateCategoryPointResult } from '@/types/pointSystem';
+
+@Component({
+  components: {
+    DialogBase: components.DialogBase,
+    TokenLogo: components.TokenLogo,
+  },
+})
+export default class TaskDialog extends Mixins(mixins.TranslationMixin, mixins.DialogMixin) {
+  @Prop({ required: true }) readonly pointsForCategory!: CalculateCategoryPointResult;
+
+  getImageSrc(imageName: string): any {
+    return getImageSrc(imageName);
+  }
+
+  get maxLevel(): number {
+    return MAX_LEVEL;
+  }
+
+  get isTokenImage(): boolean {
+    return isTokenImage(this.imageName);
+  }
+
+  private get levelCurrent(): number {
+    return this.pointsForCategory.levelCurrent;
+  }
+
+  private get imageName(): string {
+    return this.pointsForCategory.imageName;
+  }
+
+  get progressPercentage(): number {
+    if (!this.pointsForCategory.minimumAmountForNextLevel) {
```

Comment:
Please check that `minimumAmountForNextLevel ` is not zero value, to not divide on zero below